File: rust/examples/test_strings.rs
use emtrace::{Sink, expect, init, trace, traceln};

fn main() {
    expect!(
        "Hello, World!\n\
         Empty string: \n\
         Unicode: 😀 ✅ ©\n\
         Multiple strings: first second third\n"
    );

    init(&mut std::io::stdout().lock()).unwrap();

    // Test basic string
    traceln!("{}", str: "Hello, World!");

    // Test empty string
    trace!("Empty string: ");
    traceln!("{}", str: "");

    // Test unicode strings
    traceln!("Unicode: {} {} {}",
        str: "😀",
        str: "✅",
        str: "©"
    );

    // Test multiple strings
    traceln!("Multiple strings: {} {} {}",
        str: "first",
        str: "second",
        str: "third"
    );
}

File: c/examples/test_strings.c
#include "test_utils.h"
#include <emtrace/emtrace.h>

EXPECT_OUTPUT(
    "Hello, World!\n"
    "Empty string: \n"
    "Unicode: 😀✅©\n"
    "Long string: first second third\n"
);

int main(void) {
    EMTRACE_INIT();

    // Test basic string
    EMTRACELN("Hello, World!");

    // Test empty string
    EMTRACE("Empty string: ");
    EMTRACELN("");

    // Test unicode strings
    const char* emoji_smile = "😀";
    const char* emoji_check = "✅";
    const char* copyright = "©";

    EMTRACE_F("Unicode: {}{}{}\n", STR(emoji_smile), STR(emoji_check), STR(copyright));

    // Test multiple strings
    const char* long_string = "first second third";

    EMTRACE("Long string: ");
    EMTRACE_F("{}\n", STR(long_string));

    return 0;
}
